draft-kazuho-httpbis-priority.md

```diff
@@ -112,33 +112,103 @@ Unknown parameters MUST be ignored.
+| Urgency         | Definition                      |
+|----------------:|:--------------------------------|
+|              -1 | blocking ({{blocking}})         |
+|               0 | document ({{document}})         |
+| between 1 and 5 | non-blocking ({{non-blocking}}) |
+|               6 | background ({{background}})     |
+{: #urgencies title="Urgencies"}
 
-The default value is `document`.
+The value is encoded as an sh-integer.  The default value is zero.
 
-A server SHOULD transmit HTTP responses in the order of their urgency:
-`blocking` first, followed by `document`, followed by `non-blocking`.
+A server SHOULD transmit HTTP responses in the order of their urgency values.
+The lesser the value is, the higher the precedence is.
 
 The following example shows a request for a CSS file with the urgency set to
-`blocking`:
+`-1`:
 
 ~~~ example
 :method = GET
 :scheme = https
 :authority = example.net
 :path = /style.css
-priority = urgency=blocking
+priority = urgency=-1
 ~~~
 
+The definition of the urgencies and their expected use-case are described below.
+Endpoints SHOULD respect the definition of the values when assigning urgencies.
+
+### blocking
+
+The blocking urgency indicates that the response prevents other responses from
+being used.
+
+Value of -1 is used to represent this urgency.
+
+For example, use of an external stylesheet can block a web browser from
+rendering the HTML. In such case, the stylesheet is given the blocking urgency.
+
+### document
+
+The document urgency indicates that the response contains the document that is
+being processed.  This urgency is also used for responses that deserve the same
+precedence as the contents of the document.
+
+Value of zero (0) is used to represent this urgency.
+
+For example, when a user using a web browser navigates to a different HTML
+document, the request for that HTML is given the document urgency.  When that
+HTML document uses a custom font, the request for that custom font SHOULD also
+be given the document urgency.  This is because the availablity of the custom
+font is likely a precondition for the user to use that portion of the HTML
+document, which is to be rendered by that font.
+
+### non-blocking
+
+The non-blocking urgency indicates that the response does not prevent the client
+from using the document even though the response is being incorporated to or
+referred to by the document.
+
+For example, inline images (i.e., images being fetched and displayed as part of
+the document) are visually important elements of an HTML document.  As such,
+users will typically not be prevented from using the document, at least to some
+degree, before any or all of these images are loaded.  Display of those images
+are thus considered to be an improvement for visual clients rather than a
+prerequisite for all user agents.  Therefore, such images will be given the
+non-blocking urgency.
+
+Values between 1 and 5 are used to represent this urgency, to provide wiggle
```

Comment:
Did I miss where we state that `1 is high` and `5 is low`?

Reply:
We have the general rule stated; i.e., "The lower the value, the higher the precedence."

That said, I have added 0b516b9 that stresses the meanings of the values.